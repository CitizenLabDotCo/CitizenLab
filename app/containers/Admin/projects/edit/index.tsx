// Libraries
import * as React from 'react';
import * as Rx from 'rxjs/Rx';

// Services
import { projectBySlugStream, IProjectData } from 'services/projects';

// Components
import GoBackButton from 'components/UI/GoBackButton';
import TabbedResource from 'components/admin/TabbedResource';
import { browserHistory } from 'react-router';

// Localisation
import { InjectedIntlProps } from 'react-intl';
import { injectIntl } from 'utils/cl-intl';
import messages from '../messages';

// style
import styled from 'styled-components';

const StyledGoBackButton = styled(GoBackButton)`
  margin-top: 5px;
  margin-bottom: 30px;
`;

type Props = {
  params: {
    slug: string | null,
  },
  location: {
    pathname: string
  }
};

type State = {
  project: IProjectData | null,
  loaded: boolean
};

class AdminProjectEdition extends React.PureComponent<Props & InjectedIntlProps, State> {
  props$: Rx.BehaviorSubject<Props>;
  subscriptions: Rx.Subscription[];

  constructor(props: Props) {
    super(props as any);
    this.state = {
      project: null,
      loaded: false
    };
    this.props$ = new Rx.BehaviorSubject(null as any);
    this.subscriptions = [];
  }

  componentWillMount() {
    this.props$.next(this.props);

    this.subscriptions = [
      this.props$.distinctUntilChanged().switchMap((props) => {
        return (props.params && props.params.slug ? projectBySlugStream(props.params.slug).observable : Rx.Observable.of(null));
      }).subscribe((project) => {
        this.setState({
          project: (project ? project.data : null),
          loaded: true
        });
      })
    ];
  }

  componentWillReceiveProps(nextProps) {
    this.props$.next(nextProps);
  }

  componentWillUnmount() {
    this.subscriptions.forEach(subscription => subscription.unsubscribe());
  }

  getTabs = (slug: string, project: IProjectData) => {
    const baseTabsUrl = `/admin/projects/${slug}`;

    const tabs = [
      {
        label: this.props.intl.formatMessage(messages.generalTab),
        url: `${baseTabsUrl}/edit`,
      },
      {
        label: this.props.intl.formatMessage(messages.descriptionTab),
        url: `${baseTabsUrl}/description`,
      },
      {
        label: this.props.intl.formatMessage(messages.ideasTab),
        url: `${baseTabsUrl}/ideas`,
      },
      {
        label: this.props.intl.formatMessage(messages.eventsTab),
        url: `${baseTabsUrl}/events`,
      },
      {
        label: this.props.intl.formatMessage(messages.permissionsTab),
        url: `${baseTabsUrl}/permissions`,
        feature: 'private_projects',
      },
    ];

    if (project.attributes.process_type === 'timeline') {
      tabs.splice(3, 0, {
        label: this.props.intl.formatMessage(messages.phasesTab),
        url: `${baseTabsUrl}/timeline`,
      });
    }

    return tabs;
  }

  goBack = () => {
    const currentLocation = browserHistory.getCurrentLocation();
    const currentPath = currentLocation.pathname;
    const lastUrlSegment = currentPath.substr(currentPath.lastIndexOf('/') + 1);
    const newPath = currentPath.replace(lastUrlSegment, '').replace(/\/$/, '');
    const newLastUrlSegment = newPath.substr(newPath.lastIndexOf('/') + 1);

    if (newLastUrlSegment === this.props.params.slug) {
      browserHistory.push('/admin/projects');
    } else {
      browserHistory.push(newPath);
    }
  }

  render() {
    const { location } = this.props;
    const { slug } = this.props.params;
    const { project, loaded } = this.state;
    const { formatMessage } = this.props.intl;

    if (loaded) {
      const { children } = this.props;
      const childrenWithExtraProps = React.cloneElement(children as React.ReactElement<any>, { project });
      const tabbedProps = {
        location,
        resource: {
          title: project ? project.attributes.title_multiloc : formatMessage(messages.addNewProject),
          publicLink: project ? `/projects/${project.attributes.slug}` : ''
        },
        messages: {
          viewPublicResource: messages.viewPublicProject,
        },
        tabs: ((slug && project) ? this.getTabs(slug, project) : [])
      };

      return(
        <>
          <StyledGoBackButton onClick={this.goBack} />
          <TabbedResource {...tabbedProps}>
            {childrenWithExtraProps}
          </TabbedResource>
        </>
      );
    }

    return null;
  }
}

export default injectIntl<Props>(AdminProjectEdition);
