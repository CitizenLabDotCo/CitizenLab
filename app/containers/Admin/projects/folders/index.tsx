// Libraries
import React, { PureComponent } from 'react';
import clHistory from 'utils/cl-router/history';
import { adopt } from 'react-adopt';
import { withRouter, WithRouterProps } from 'react-router';

// Utils
import { isNilOrError } from 'utils/helperUtils';

// Components
import GoBackButton from 'components/UI/GoBackButton';
import TabbedResource from 'components/admin/TabbedResource';
import Button from 'components/UI/Button';

// Localisation
import { InjectedIntlProps } from 'react-intl';
import { injectIntl, FormattedMessage } from 'utils/cl-intl';
import injectLocalize, { InjectedLocalized } from 'utils/localize';
import messages from './messages';

// Resources
import GetProjectFolder, {
  GetProjectFolderChildProps,
} from 'resources/GetProjectFolder';

// style
import styled from 'styled-components';

const TopContainer = styled.div`
  width: 100%;
  margin-top: -5px;
  margin-bottom: 30px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: relative;
`;

export interface InputProps {}

interface DataProps {
  projectFolder: GetProjectFolderChildProps;
}

interface State {}

export interface Props extends InputProps, DataProps {}

export class AdminProjectFolderEdition extends PureComponent<
  Props & InjectedIntlProps & InjectedLocalized & WithRouterProps,
  State
> {
  goBack = () => {
    clHistory.push('/admin/projects');
  };

  render() {
    const { projectFolderId } = this.props.params;
    const {
      intl: { formatMessage },
      localize,
      projectFolder,
      children,
    } = this.props;
    const tabbedProps = {
      resource: {
        title: !isNilOrError(projectFolder)
          ? localize(projectFolder.attributes.title_multiloc)
          : '',
      },
      tabs: [
        {
          label: formatMessage(messages.projectFolderProjectsTab),
          url: `/admin/projects/folders/${projectFolderId}`,
        },
        {
          label: formatMessage(messages.projectFolderSettingsTab),
          url: `/admin/projects/folders/${projectFolderId}/settings`,
        },
      ],
    };

    return (
      <>
        <TopContainer>
          <GoBackButton onClick={this.goBack} />
          {!isNilOrError(projectFolder) && (
            <Button
              buttonStyle="cl-blue"
              icon="eye"
              id="to-projectFolder"
              linkTo={`/folders/${projectFolder.attributes.slug}`}
            >
              <FormattedMessage {...messages.viewPublicProjectFolder} />
            </Button>
          )}
        </TopContainer>
        <TabbedResource {...tabbedProps}>{children}</TabbedResource>
      </>
    );
  }
}

const AdminProjectFolderEditionWithHoCs = withRouter(
  injectIntl<Props & WithRouterProps>(injectLocalize(AdminProjectFolderEdition))
);

const Data = adopt<DataProps, InputProps & WithRouterProps>({
  projectFolder: ({ params, render }) => (
    <GetProjectFolder projectFolderId={params.projectFolderId}>
      {render}
    </GetProjectFolder>
  ),
});

export default (inputProps: InputProps & WithRouterProps) => (
  <Data {...inputProps}>
    {(dataProps) => (
      <AdminProjectFolderEditionWithHoCs {...inputProps} {...dataProps} />
    )}
  </Data>
);
