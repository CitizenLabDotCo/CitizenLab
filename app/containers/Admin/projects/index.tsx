import * as React from 'react';

// components
import HelmetIntl from 'components/HelmetIntl';

// i18n
import messages from './messages';

type Props = {};

type State = {};

export default class ProjectDashboard extends React.PureComponent<Props, State> {
  render() {
    return (
      <>
        <HelmetIntl title={messages.helmetTitle} description={messages.helmetDescription} />
        {this.props.children}
      </>
    );
  }
}
