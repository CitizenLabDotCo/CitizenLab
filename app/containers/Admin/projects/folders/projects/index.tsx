import React, { Component } from 'react';
import { adopt } from 'react-adopt';
import { isNilOrError } from 'utils/helperUtils';
import { withRouter, WithRouterProps } from 'react-router';

// services
import { updateProjectFolderMembership, PublicationStatus } from 'services/projects';

// resources
import GetProjectFolder, { GetProjectFolderChildProps } from 'resources/GetProjectFolder';
import GetAdminPublications, { GetAdminPublicationsChildProps } from 'resources/GetAdminPublications';

// localisation
import { FormattedMessage } from 'utils/cl-intl';
import messages from '../messages';

// components
import { List, Row, SortableList, SortableRow } from 'components/admin/ResourceList';
import { HeaderTitle } from '../../all/StyledComponents';
import ProjectRow from '../../components/ProjectRow';

// style
import styled from 'styled-components';
import { reorderAdminPublication } from 'services/adminPublications';
import { IAdminPublicationContent } from 'hooks/useAdminPublications';

const Container = styled.div`
  min-height: 60vh;
`;

const ListsContainer = styled.div``;

const ListHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  margin-bottom: 25px;
  &:not(:first-child) {
    margin-top: 70px;
  }
  & + & {
    margin-top: 30px;
  }
`;

const StyledHeaderTitle = styled(HeaderTitle)`
  font-weight: bold;
`;

const Spacer = styled.div`
  flex: 1;
`;

interface DataProps {
  topLevelProjects: GetAdminPublicationsChildProps;
  projectFolder: GetProjectFolderChildProps;
  projectsInFolder: GetAdminPublicationsChildProps;
}

interface Props extends DataProps { }

interface State {
  processing: string[];
}

class AdminFoldersProjectsList extends Component<Props & WithRouterProps, State> {
  constructor(props) {
    super(props);

    this.state = {
      processing: []
    };
  }

  handleReorder = (itemId, newOrder) => {
    reorderAdminPublication(itemId, newOrder);
  }

  addProjectToFolder = (projectId: string) => async () => {
    const projectFolderId = !isNilOrError(this.props.projectFolder) ? this.props.projectFolder.id : null;
    if (projectFolderId) {
      this.setState(({ processing }) => ({ processing: [...processing, projectId] }));
      await updateProjectFolderMembership(projectId, projectFolderId);
      this.setState(({ processing }) => ({ processing: processing.filter(item => item !== projectId) }));
    }
  }

  removeProjectFromFolder = (projectId: string) => async () => {
    const projectFolderId = !isNilOrError(this.props.projectFolder) ? this.props.projectFolder.id : undefined;
    this.setState(({ processing }) => ({ processing: [...processing, projectId] }));
    await updateProjectFolderMembership(projectId, null, projectFolderId);
    this.setState(({ processing }) => ({ processing: processing.filter(item => item !== projectId) }));
  }

  render() {
    const { topLevelProjects, projectsInFolder } = this.props;

    const { processing } = this.state;

    const otherProjects = (!isNilOrError(topLevelProjects) && topLevelProjects.list)
      ? topLevelProjects.list.filter(item => item.publicationType === 'project')
      : null;

    const inFolderFinalList = (!isNilOrError(projectsInFolder) && projectsInFolder.list && projectsInFolder.list.length > 0)
      ? projectsInFolder.list.filter(item => item.publicationType === 'project')
      : null;

    return (
      <Container>
        <ListsContainer>
          <ListHeader>
            <StyledHeaderTitle>
              <FormattedMessage {...messages.projectsAlreadyAdded} />
            </StyledHeaderTitle>

            <Spacer />

          </ListHeader>

          {inFolderFinalList ?
            <SortableList
              key={`IN_FOLDER_LIST${inFolderFinalList.length}`}
              items={inFolderFinalList}
              onReorder={this.handleReorder}
              className="projects-list e2e-admin-folder-projects-list"
              id="e2e-admin-fodlers-projects-list"
            >
              {({ itemsList, handleDragRow, handleDropRow }) => (
                itemsList.map((adminPublication: IAdminPublicationContent, index) => {
                  return (
                    <SortableRow
                      key={adminPublication.id}
                      id={adminPublication.id}
                      index={index}
                      moveRow={handleDragRow}
                      dropRow={handleDropRow}
                      lastItem={(index === itemsList.length - 1)}
                    >
                      <ProjectRow
                        publication={adminPublication}
                        actions={[{
                          buttonContent: <FormattedMessage {...messages.removeFromFolder} />,
                          handler: this.removeProjectFromFolder,
                          icon: 'remove',
                          processing: processing.includes(adminPublication.publicationId)
                        }, 'manage']}
                      />
                    </SortableRow>
                  );
                })
              )}
            </SortableList>
            :
            <FormattedMessage {...messages.emptyFolder} />
          }
          <ListHeader>
            <StyledHeaderTitle>
              <FormattedMessage {...messages.projectsYouCanAdd} />
            </StyledHeaderTitle>
          </ListHeader>

          {otherProjects ?
            <List key={`JUST_LIST${otherProjects.length}`}>
              {otherProjects.map((adminPublication, index: number) => {
                return (
                  <Row
                    id={adminPublication.id}
                    lastItem={(index === otherProjects.length - 1)}
                    key={adminPublication.id}
                  >
                    <ProjectRow
                      publication={adminPublication}
                      actions={[{
                        buttonContent: <FormattedMessage {...messages.addToFolder} />,
                        handler: this.addProjectToFolder,
                        processing: processing.includes(adminPublication.publicationId),
                        icon: 'plus-circle'
                      }]}
                    />
                  </Row>
                );
              })}
            </List>
            : <FormattedMessage {...messages.noProjectsToAdd} />
          }
        </ListsContainer>
      </Container>
    );
  }
}
const AdminFoldersProjectsListWithHocs = withRouter(AdminFoldersProjectsList);

const publicationStatuses: PublicationStatus[] = ['draft', 'archived', 'published'];

const Data = adopt<DataProps, WithRouterProps>({
  projectFolder: ({ params, render }) => <GetProjectFolder projectFolderId={params.projectFolderId}>{render}</GetProjectFolder>,
  topLevelProjects: <GetAdminPublications publicationStatusFilter={publicationStatuses} folderId={null} />,
  projectsInFolder: ({ params, render }) => <GetAdminPublications publicationStatusFilter={publicationStatuses} folderId={params.projectFolderId}>{render}</GetAdminPublications>,
});

export default (inputProps: WithRouterProps) => (
  <Data {...inputProps}>
    {dataProps => <AdminFoldersProjectsListWithHocs {...inputProps} {...dataProps} />}
  </Data>
);
