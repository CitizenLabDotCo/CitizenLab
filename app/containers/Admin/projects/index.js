import React from 'react';
import PropTypes from 'prop-types';
import HelmetIntl from 'components/HelmetIntl';

// components
// import { FormattedMessage } from 'react-intl';
import WatchSagas from 'utils/containers/watchSagas';
import sagas from 'resources/projects/sagas';

// store
import { preprocess } from 'utils';

// messages
import messages from './messages';
import { resetProjects } from 'resources/projects/actions';

class ProjectDashboard extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function

  componentWillUnmount() {
    // reset projects upon page leaving, to avoid double items when going back
    this.props.resetProjects();
  }

  render() {
    return (
      <div>
        <HelmetIntl
          title={messages.helmetTitle}
          description={messages.helmetDescription}
        />
        <WatchSagas sagas={sagas} />
        {this.props.children}
      </div>

    );
  }
}

ProjectDashboard.propTypes = {
  resetProjects: PropTypes.func.isRequired,
  children: PropTypes.element,
};

export default preprocess(null, { resetProjects })(ProjectDashboard);
