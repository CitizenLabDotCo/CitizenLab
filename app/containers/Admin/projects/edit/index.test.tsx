// libraries
import React from 'react';

// component to test
// ! it is the component without the HoCs (data, localization...)
import { AdminProjectEdition } from './';

// mock data and functions to replace the HoCs
import { GetPhasesChildProps } from 'resources/GetPhases';
import {
  mockPhaseIdeationData,
  mockPhaseSurveyTypeformData,
  mockPhaseSurveyGoogleFormData,
  mockPhaseInformationData
} from 'services/__mocks__/phases.ts';
import { getMockProject } from 'services/__mocks__/projects.ts';
import { shallowWithIntl } from 'utils/testUtils/withIntl';
import { localizeProps } from 'utils/testUtils/localizeProps';
import { WithRouterProps } from 'react-router';

// what needs to be mocked by jest to render the component
jest.mock('utils/cl-intl');

const getRouterProps = (projectId, tabName?: string) => ({
  location: {
    pathname: `/admin/projects/${projectId}/${tabName}`,
  },
  params: {
    projectId
  }
}) as any as WithRouterProps;

const children = () => <div/>;

describe('<AdminProjectEdition />', () => {

  it('renders the correct set of tabs for a continuous non-survey non-ideation project', () => {
    const surveys_enabled = true;
    const typeform_enabled = true;
    const phases = [];
    const project = getMockProject('continuousInformation', 'continuous', 'information');
    const routerProps = getRouterProps('continuousInformation');

    // passing in mock data to replace the HoCs
    // ignore the error, shallowWithIntl method passes in the intl object
    const wrapper = shallowWithIntl((
      <AdminProjectEdition
        surveys_enabled={surveys_enabled}
        typeform_enabled={typeform_enabled}
        phases={phases}
        project={project}
        children={children}
        {...routerProps}
        {...localizeProps}
      />));
      const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

    expect(tabs.map(tab => tab.url)).toMatchSnapshot();
  });

  it('renders the correct set of tabs for a continuous typeform survey project', () => {
    const surveys_enabled = true;
    const typeform_enabled = true;
    const phases = [];
    const project = getMockProject('continuousTypeform', 'continuous', 'survey', 'typeform');
    const routerProps = getRouterProps('continuousTypeform');

    // passing in mock data to replace the HoCs
    // ignore the error, shallowWithIntl method passes in the intl object
    const wrapper = shallowWithIntl((
      <AdminProjectEdition
        surveys_enabled={surveys_enabled}
        typeform_enabled={typeform_enabled}
        phases={phases}
        project={project}
        children={children}
        {...routerProps}
        {...localizeProps}
      />));
      const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

    expect(tabs.map(tab => tab.url)).toMatchSnapshot();
  });

  it('renders the correct set of tabs for a continuous non-typeform survey project', () => {
    const surveys_enabled = true;
    const typeform_enabled = true;
    const phases = [];
    const project = getMockProject('continuousGoogleForm', 'continuous', 'survey', 'google_forms');
    const routerProps = getRouterProps('continuousGoogleForm');

    // passing in mock data to replace the HoCs
    // ignore the error, shallowWithIntl method passes in the intl object
    const wrapper = shallowWithIntl((
      <AdminProjectEdition
        surveys_enabled={surveys_enabled}
        typeform_enabled={typeform_enabled}
        phases={phases}
        project={project}
        children={children}
        {...routerProps}
        {...localizeProps}
      />));
      const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

    expect(tabs.map(tab => tab.url)).toMatchSnapshot();
  });

  it('renders the correct set of tabs for a continuous typeform survey project if surveys are disabled', () => {
    const surveys_enabled = false;
    const typeform_enabled = true;
    const phases = [];
    const project = getMockProject('continuousTypeform', 'continuous', 'survey', 'typeform');
    const routerProps = getRouterProps('continuousTypeform');

    // passing in mock data to replace the HoCs
    // ignore the error, shallowWithIntl method passes in the intl object
    const wrapper = shallowWithIntl((
      <AdminProjectEdition
        surveys_enabled={surveys_enabled}
        typeform_enabled={typeform_enabled}
        phases={phases}
        project={project}
        children={children}
        {...routerProps}
        {...localizeProps}
      />));
      const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

      expect(tabs.map(tab => tab.url)).toMatchSnapshot();
    });

  it('renders the correct set of tabs for a continuous typeform survey project if typeform is disabled', () => {
    const surveys_enabled = true;
    const typeform_enabled = false;
    const phases = [];
    const project = getMockProject('continuousTypeform', 'continuous', 'survey', 'typeform');
    const routerProps = getRouterProps('continuousTypeform');

    // passing in mock data to replace the HoCs
    // ignore the error, shallowWithIntl method passes in the intl object
    const wrapper = shallowWithIntl((
      <AdminProjectEdition
        surveys_enabled={surveys_enabled}
        typeform_enabled={typeform_enabled}
        phases={phases}
        project={project}
        children={children}
        {...routerProps}
        {...localizeProps}
      />));
      const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

      expect(tabs.map(tab => tab.url)).toMatchSnapshot();
    });

    it('renders the correct set of tabs for a timeline project without ideation nor surveys', () => {
      const surveys_enabled = true;
      const typeform_enabled = true;
      const project = getMockProject('timelineInformation', 'timeline');
      const phases = [mockPhaseInformationData];
      const routerProps = getRouterProps('timelineInformation');
      // weirdly enough, project without ideation still have an ideas tab

      // passing in mock data to replace the HoCs
      // ignore the error, shallowWithIntl method passes in the intl object
      const wrapper = shallowWithIntl((
        <AdminProjectEdition
          surveys_enabled={surveys_enabled}
          typeform_enabled={typeform_enabled}
          phases={phases}
          project={project}
          children={children}
          {...routerProps}
          {...localizeProps}
        />));
        const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

      expect(tabs.map(tab => tab.url)).toMatchSnapshot();
    });

    it('renders the correct set of tabs for a timeline project with an ideation phase', () => {
      const surveys_enabled = true;
      const typeform_enabled = true;
      const project = getMockProject('timelineIdeation', 'timeline');
      const phases = [mockPhaseInformationData, mockPhaseIdeationData] as GetPhasesChildProps;
      const routerProps = getRouterProps('timelineIdeation');

      // passing in mock data to replace the HoCs
      // ignore the error, shallowWithIntl method passes in the intl object
      const wrapper = shallowWithIntl((
        <AdminProjectEdition
          surveys_enabled={surveys_enabled}
          typeform_enabled={typeform_enabled}
          phases={phases}
          project={project}
          children={children}
          {...routerProps}
          {...localizeProps}
        />));
        const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

      expect(tabs.map(tab => tab.url)).toMatchSnapshot();
    });

    it('renders the correct set of tabs for a timeline project with a typeform survey phase', () => {
      const surveys_enabled = true;
      const typeform_enabled = true;
      const project = getMockProject('timelineIdeation', 'timeline');
      const phases = [mockPhaseInformationData, mockPhaseSurveyTypeformData];
      const routerProps = getRouterProps('timelineIdeation');

      // passing in mock data to replace the HoCs
      // ignore the error, shallowWithIntl method passes in the intl object
      const wrapper = shallowWithIntl((
        <AdminProjectEdition
          surveys_enabled={surveys_enabled}
          typeform_enabled={typeform_enabled}
          phases={phases}
          project={project}
          children={children}
          {...routerProps}
          {...localizeProps}
        />));
        const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

      expect(tabs.map(tab => tab.url)).toMatchSnapshot();
    });

    it('renders the correct set of tabs for a timeline project with a googleform survey phase', () => {
      const surveys_enabled = true;
      const typeform_enabled = true;
      const project = getMockProject('timelineIdeation', 'timeline');
      const phases = [mockPhaseInformationData, mockPhaseSurveyGoogleFormData];
      const routerProps = getRouterProps('timelineIdeation');

      // passing in mock data to replace the HoCs
      // ignore the error, shallowWithIntl method passes in the intl object
      const wrapper = shallowWithIntl((
        <AdminProjectEdition
          surveys_enabled={surveys_enabled}
          typeform_enabled={typeform_enabled}
          phases={phases}
          project={project}
          children={children}
          {...routerProps}
          {...localizeProps}
        />));
        const tabs = wrapper.find('withRouter(TabbedResource)').props().tabs;

      expect(tabs.map(tab => tab.url)).toMatchSnapshot();
    });

    it('lets you see the project', () => {
      const surveys_enabled = true;
      const typeform_enabled = true;
      const phases = [];
      const project = getMockProject('continuousInformation', 'continuous', 'information');
      const routerProps = getRouterProps('continuousInformation');

      // passing in mock data to replace the HoCs
      // ignore the error, shallowWithIntl method passes in the intl object
      const wrapper = shallowWithIntl((
        <AdminProjectEdition
          surveys_enabled={surveys_enabled}
          typeform_enabled={typeform_enabled}
          phases={phases}
          project={project}
          children={children}
          {...routerProps}
          {...localizeProps}
        />));
        const linkPath = wrapper.find('#to-project').props().linkTo;

      expect(linkPath).toMatchSnapshot();
    });

    it('lets you add an idea when on the ideas tab', () => {
      const surveys_enabled = true;
      const typeform_enabled = true;
      const phases = [];
      const project = getMockProject('continuousIdeation', 'continuous', 'ideation');
      const routerProps = getRouterProps('continuousIdeation', 'ideas');

      // passing in mock data to replace the HoCs
      // ignore the error, shallowWithIntl method passes in the intl object
      const wrapper = shallowWithIntl((
        <AdminProjectEdition
          surveys_enabled={surveys_enabled}
          typeform_enabled={typeform_enabled}
          phases={phases}
          project={project}
          children={children}
          {...routerProps}
          {...localizeProps}
        />));
        const linkPath = wrapper.find('#new-idea').props().linkTo;

      expect(linkPath).toMatchSnapshot();
    });

    it('doesn\'t you add an idea when there\'s no ideation phase', () => {
      const surveys_enabled = true;
      const typeform_enabled = true;
      const phases = [];
      const project = getMockProject('continuousInformation', 'continuous', 'information');
      const routerProps = getRouterProps('continuousInformation', 'general');

      // passing in mock data to replace the HoCs
      // ignore the error, shallowWithIntl method passes in the intl object
      const wrapper = shallowWithIntl((
        <AdminProjectEdition
          surveys_enabled={surveys_enabled}
          typeform_enabled={typeform_enabled}
          phases={phases}
          project={project}
          children={children}
          {...routerProps}
          {...localizeProps}
        />));
        const linkPath = wrapper.find('#new-idea');

      expect(linkPath).toHaveLength(0);
    });
});
