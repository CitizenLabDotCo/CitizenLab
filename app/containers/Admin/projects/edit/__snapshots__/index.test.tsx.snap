// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminProjectEdition /> renders the correct set of tabs for a continuous non-survey non-ideation project 1`] = `
Array [
  "/admin/projects/continuousInformation/edit",
  "/admin/projects/continuousInformation/description",
  "/admin/projects/continuousInformation/events",
  "/admin/projects/continuousInformation/permissions",
]
`;

exports[`<AdminProjectEdition /> renders the correct set of tabs for a continuous non-typeform survey project 1`] = `
Array [
  "/admin/projects/continuousGoogleForm/edit",
  "/admin/projects/continuousGoogleForm/description",
  "/admin/projects/continuousGoogleForm/events",
  "/admin/projects/continuousGoogleForm/permissions",
]
`;

exports[`<AdminProjectEdition /> renders the correct set of tabs for a continuous typeform survey project 1`] = `
Array [
  "/admin/projects/continuousTypeform/edit",
  "/admin/projects/continuousTypeform/description",
  "/admin/projects/continuousTypeform/events",
  "/admin/projects/continuousTypeform/survey-results",
  "/admin/projects/continuousTypeform/permissions",
]
`;

exports[`<AdminProjectEdition /> renders the correct set of tabs for a continuous typeform survey project if surveys are disabled 1`] = `
Array [
  "/admin/projects/continuousTypeform/edit",
  "/admin/projects/continuousTypeform/description",
  "/admin/projects/continuousTypeform/events",
  "/admin/projects/continuousTypeform/permissions",
]
`;

exports[`<AdminProjectEdition /> renders the correct set of tabs for a continuous typeform survey project if typeform is disabled 1`] = `
Array [
  "/admin/projects/continuousTypeform/edit",
  "/admin/projects/continuousTypeform/description",
  "/admin/projects/continuousTypeform/events",
  "/admin/projects/continuousTypeform/permissions",
]
`;

exports[`<AdminProjectEdition /> renders the correct set of tabs for a timeline project with a googleform survey phase 1`] = `
Array [
  "/admin/projects/timelineIdeation/edit",
  "/admin/projects/timelineIdeation/description",
  "/admin/projects/timelineIdeation/ideas",
  "/admin/projects/timelineIdeation/timeline",
  "/admin/projects/timelineIdeation/poll",
  "/admin/projects/timelineIdeation/events",
  "/admin/projects/timelineIdeation/permissions",
]
`;

exports[`<AdminProjectEdition /> renders the correct set of tabs for a timeline project with a typeform survey phase 1`] = `
Array [
  "/admin/projects/timelineIdeation/edit",
  "/admin/projects/timelineIdeation/description",
  "/admin/projects/timelineIdeation/ideas",
  "/admin/projects/timelineIdeation/survey-results",
  "/admin/projects/timelineIdeation/timeline",
  "/admin/projects/timelineIdeation/poll",
  "/admin/projects/timelineIdeation/events",
  "/admin/projects/timelineIdeation/permissions",
]
`;

exports[`<AdminProjectEdition /> renders the correct set of tabs for a timeline project with an ideation phase 1`] = `
Array [
  "/admin/projects/timelineIdeation/edit",
  "/admin/projects/timelineIdeation/description",
  "/admin/projects/timelineIdeation/ideas",
  "/admin/projects/timelineIdeation/timeline",
  "/admin/projects/timelineIdeation/poll",
  "/admin/projects/timelineIdeation/events",
  "/admin/projects/timelineIdeation/permissions",
]
`;

exports[`<AdminProjectEdition /> renders the correct set of tabs for a timeline project without ideation nor surveys 1`] = `
Array [
  "/admin/projects/timelineInformation/edit",
  "/admin/projects/timelineInformation/description",
  "/admin/projects/timelineInformation/ideas",
  "/admin/projects/timelineInformation/timeline",
  "/admin/projects/timelineInformation/poll",
  "/admin/projects/timelineInformation/events",
  "/admin/projects/timelineInformation/permissions",
]
`;
